### PR TITLE
ci: update build and deployment workflows

### LINKED ISSUE #203: [BUG]: fix building the terarium applications
**Describe the bug**
The build scripts are not working for terarium right now

**To Reproduce**

**Expected behavior**
the hmi-server should be built in both JVM and Native mode and the webapp should be built

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by Nithos**
# Description

Update the way TERArium is built and deployed using the bake file. In theory it is setup to trigger once the lint and tests pass and finish, however restricting to just a tag is undocumented. So this will need to be tested still but unfortunately will need to be merged first :(

Resolves #203 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Workflows are manual tested within draft mode. Deployment tests are done manually and removed there after.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by YohannParis — DISMISSED** [review]

File: .github/workflows/playwright.yml [hunk 1]
```diff
@@ -4,10 +4,15 @@
 
 name: Playwright Tests
 on:
-  push:
-    branches: [main]
-  pull_request:
-    branches: [main]
+  # push:
```
Comment (dgauldie) [inline .github/workflows/playwright.yml:7]: do we need this anymore?

Comment (dgauldie) [reply .github/workflows/playwright.yml:7]: I'm happy to do a clean-up once we have the CI working exactly the way we want

File: Makefile [hunk 2]
```diff
@@ -1,9 +1,7 @@
 version=3.81
 
 PROJECT_DIR ?= .
-DOCKER_IMAGE_TAG ?= dev
-DOCKER_REGISTRY := ghcr.io
-DOCKER_ORG := darpa-askem
+PACKAGE_TYPE := jar
```
Comment (dgauldie) [inline Makefile:4]: we don't need this anymore as we hardcode the build scripts

File: Makefile [hunk 3]
```diff
@@ -17,53 +15,48 @@ all:
...
-	@echo "  publish                  - build and push dock images for deployment"
-	@echo "  publish-<target>         - build and push dock image for a specific target - see below for list of targets"
-	@echo "  lint                     - lint the source code"
-	@echo "  test                     - test the source code"
-	@echo "Targets building (but not pushing) docker images of separately:"
-	@echo "  $(TARGETS)"
 	@echo ""
 	@echo "Variables:"
-	@echo "  DOCKER_IMAGE_TAG=<tag>         - tag to append to each pushed docker image (default: dev)"
+	@echo "  PACKAGE_TYPE=<type>      - possible hmi-server package types are: `jar` or `native` (default: jar)"
```
Comment (dgauldie) [inline Makefile:20]: we don't need this either. We could remove the `Variables` section altogether

**Comment by dgauldie** [comment]
finished first pass

File: Makefile (hunk 2 above)
Comment (dgauldie) [reply Makefile:4]: fixed

File: Makefile (hunk 3 above)
Comment (dgauldie) [reply Makefile:20]: fixed

File: docker-bake.hcl [hunk 4]
```diff
@@ -7,38 +7,59 @@ variable "DOCKER_ORG" {
...
+  targets = ["hmi-client-base", "hmi-server-base"]
 }
 
 # ---------------------------------
 target "_platforms" {
   platforms = ["linux/amd64", "linux/arm64"]
+}
+
+target "hmi-client-base" {
+	context = "packages/client/webapp/docker"
```
[the remark this answers is not in the document]
Comment (dgauldie) [reply docker-bake.hcl:42]: fixed

**Review by dgauldie — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -12,15 +12,23 @@ To develop and test the core application:
...
-- `kubectl scale --replicas=0 -f webapp-deployment.yaml`
+- `kubectl delete -f [NAME]-deployment.yaml -f [NAME]-service.yaml`
 
-3. Start up the Backend Quarkus server using `./gradlew quarkusDev` or `quarkus dev` command
-4. Start the front end by doing `yarn install` followed by `yarn && yarn dev`
-5. Navigate your browser to `localhost:8078`
+3. Start a service locally to replace the services you shut down
+
+- Start up the Backend Quarkus server using `./gradlew quarkusDev` or `quarkus dev` command
+- Start the front end by doing `yarn install` followed by `yarn && yarn dev`
```
Comment (YohannParis) [inline .github/CONTRIBUTING.md:24]:
```suggestion
- Start the front end by doing `yarn && yarn dev` (shortcut for `yarn install && yarn workspaces him-client dev`)
```

File: README.md [hunk 6]
```diff
@@ -27,48 +34,55 @@ brew install quarkusio/tap/quarkus
 
 ## Running the server in dev mode
 
-First, launch the authentication service before running the server (see documentation [here](kubernetes/local/README.md)).
+First, launch the authentication service before running the server (see
+documentation [here](kubernetes/local/README.md)).
```
Comment (YohannParis) [inline README.md:38]:
```suggestion
documentation [here](https://github.com/DARPA-ASKEM/orchestration)).
```

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (dgauldie) [reply .github/CONTRIBUTING.md]: fixed

File: README.md (hunk 6 above)
Comment (dgauldie) [reply README.md]: fixed

**Review by YohannParis — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (YohannParis) [reply README.md]: In the future, for this kind of correction, do you guys prefer that I submit a commit directly?
When it's just obvious simple information, like documentation, etc.

**Review by YohannParis — APPROVED** [review]

**Pull request merged**